src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspaceManager.cs

```diff
@@ -68,12 +68,26 @@ private static RemoteWorkspace CreatePrimaryWorkspace()
         public virtual RemoteWorkspace GetWorkspace()
             => _lazyPrimaryWorkspace.Value;
 
-        public ValueTask<Solution> GetSolutionAsync(ServiceBrokerClient client, PinnedSolutionInfo solutionInfo, CancellationToken cancellationToken)
+        /// <summary>
+        /// Not ideal that we exposing the workspace solution, while not ensuring it stays alive for other calls using
+        /// the same <see cref="PinnedSolutionInfo.SolutionChecksum"/>). However, this is used by
+        /// Pythia/Razor/UnitTesting which all assume they can get that solution instance and use as desired by them.
+        /// </summary>
+        public async ValueTask<Solution> GetSolutionAsync(ServiceBrokerClient client, PinnedSolutionInfo solutionInfo, CancellationToken cancellationToken)
```

Comment:
>GetSolutionAsync [](http://example.com/codeflow?start=41&length=16)

Add new API that can be used by partners to migrate away from this one?

Reply:
sure.